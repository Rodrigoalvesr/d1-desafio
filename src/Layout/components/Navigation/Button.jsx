import React from 'react';
import PropTypes from 'prop-types';
import Tooltip from '@Components/Tooltip';
import Icon from '@Components/Icon';
import { makeStyles } from '@material-ui/core/styles';
import classnames from 'classnames';

const useStyles = makeStyles((theme) => ({
  button: {
    display: 'flex',
    justifyContent: 'center',
  },
  content: {
    width: 21,
    padding: '24px 0',
    '&.selected': {
      borderWidth: '0 0 1px 0',
      borderStyle: 'solid',
      borderImage: 'linear-gradient(90deg, #00E1FF 0%, #117EFF 100%) 1',
      boxSizing: 'border-box',
    },
  },
  icon: {
    cursor: 'pointer',
    fill: '#E4E6F1',
    '&:hover': {
      fill: theme.palette.primary.main,
    },
  },
}));

function Button({ name, icon, selected }) {
  const classes = useStyles();

  return (
    <div className={classes.button}>
      <div className={classnames(classes.content, { selected })}>
        <Tooltip title={name} placement="right" arrow>
          <div>
            <Icon name={icon} className={classes.icon} />
          </div>
        </Tooltip>
      </div>
    </div>
  );
}

Button.defaultProps = {
  selected: false,
};

Button.propTypes = {
  name: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  selected: PropTypes.bool,
};

export default Button;
