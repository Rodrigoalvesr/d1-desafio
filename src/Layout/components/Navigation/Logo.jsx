import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import logo from '@/assets/images/logotipo.png';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    justifyContent: 'center',
    marginTop: 35,
    marginBottom: 35,
  },
});

function Logo() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <a href="https://www.d1.cx/">
        <img alt="D1" src={logo} height={22} />
      </a>
    </div>
  );
}

export default Logo;
