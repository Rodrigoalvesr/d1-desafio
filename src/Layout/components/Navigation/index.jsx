import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Logo from './Logo';
import Button from './Button';

const useStyles = makeStyles({
  root: {
    position: 'relative',
    minWidth: 71,
  },
  navigation: {
    display: 'flex',
    flexDirection: 'column',
    position: 'fixed',
    height: '100vh',
    minWidth: 71,
    background: '#1A1731',
    borderWidth: '3px 0 3px 0',
    borderStyle: 'solid',
    borderImage: 'linear-gradient(90deg, #00E1FF 0%, #117EFF 100%) 1',
    boxSizing: 'border-box',
  },
  content: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
    marginTop: 18,
    marginBottom: 18,
  },
});

function Navigation() {
  const classes = useStyles();

  const createNav = (name, icon, selected = false) => {
    return { name, icon, selected };
  };

  const systemNavigation = [
    createNav('Análises', 'chart'),
    createNav('Jornadas', 'rocket'),
    createNav('Clientes', 'users'),
    createNav('CCM Cloud', 'cloud', true),
    createNav('Versão 01', 'link'),
  ];

  const userNavigation = [
    createNav('Administração', 'gem'),
    createNav('Help Desk', 'tools'),
    createNav('Trocar Conta', 'exchange'),
    createNav('Sair', 'logout'),
  ];

  return (
    <div className={classes.root}>
      <nav className={classes.navigation}>
        <Logo />

        <div className={classes.content}>
          <div>
            {systemNavigation.map((item) => (
              <Button key={`system_nav_${item.icon}`} {...item} />
            ))}
          </div>

          <div>
            {userNavigation.map((item) => (
              <Button key={`user_nav_${item.icon}`} {...item} />
            ))}
          </div>
        </div>
      </nav>
    </div>
  );
}

export default Navigation;
