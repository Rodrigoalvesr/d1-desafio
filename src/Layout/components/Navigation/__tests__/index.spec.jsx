import React from 'react';
import renderer from 'react-test-renderer';
import Navigation, { createNav } from '../index';

describe('Navigation', () => {
  it('Renders correctly', () => {
    const component = renderer.create(<Navigation />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Should return an object with icon name and selected false', () => {
    const status = createNav('An치lises', 'chart');
    const mock = {
      name: 'An치lises',
      icon: 'chart',
      selected: false,
    };
    expect(status).toEqual(mock);
  });

  it('Should return an object with icon name and selected true', () => {
    const status = createNav('An치lises', 'chart', true);
    const mock = {
      name: 'An치lises',
      icon: 'chart',
      selected: true,
    };
    expect(status).toEqual(mock);
  });
});
