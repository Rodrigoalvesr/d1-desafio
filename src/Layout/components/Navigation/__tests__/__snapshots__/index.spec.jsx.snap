// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <nav
    className="makeStyles-navigation-2"
  >
    <div
      className="makeStyles-root-4"
    >
      <a
        href="https://www.d1.cx/"
      >
        <img
          alt="D1"
          height={22}
          src="logotipo.png"
        />
      </a>
    </div>
    <div
      className="makeStyles-content-3"
    >
      <div>
        <div
          className="makeStyles-button-5"
        >
          <div
            className="makeStyles-content-6"
          >
            <div
              aria-describedby={null}
              className=""
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              title="Análises"
            >
              <svg
                className="makeStyles-root-10 makeStyles-root-11 makeStyles-icon-7"
              >
                chart-pie.svg
              </svg>
            </div>
          </div>
        </div>
        <div
          className="makeStyles-button-5"
        >
          <div
            className="makeStyles-content-6"
          >
            <div
              aria-describedby={null}
              className=""
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              title="Jornadas"
            >
              <svg
                className="makeStyles-root-10 makeStyles-root-12 makeStyles-icon-7"
              >
                rocket.svg
              </svg>
            </div>
          </div>
        </div>
        <div
          className="makeStyles-button-5"
        >
          <div
            className="makeStyles-content-6"
          >
            <div
              aria-describedby={null}
              className=""
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              title="Clientes"
            >
              <svg
                className="makeStyles-root-10 makeStyles-root-13 makeStyles-icon-7"
              >
                user-friends.svg
              </svg>
            </div>
          </div>
        </div>
        <div
          className="makeStyles-button-5"
        >
          <div
            className="makeStyles-content-6 selected"
          >
            <div
              aria-describedby={null}
              className=""
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              title="CCM Cloud"
            >
              <svg
                className="makeStyles-root-10 makeStyles-root-14 makeStyles-icon-7"
              >
                ccm-cloud.svg
              </svg>
            </div>
          </div>
        </div>
        <div
          className="makeStyles-button-5"
        >
          <div
            className="makeStyles-content-6"
          >
            <div
              aria-describedby={null}
              className=""
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              title="Versão 01"
            >
              <svg
                className="makeStyles-root-10 makeStyles-root-15 makeStyles-icon-7"
              >
                external-link-square-alt.svg
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="makeStyles-button-5"
        >
          <div
            className="makeStyles-content-6"
          >
            <div
              aria-describedby={null}
              className=""
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              title="Administração"
            >
              <svg
                className="makeStyles-root-10 makeStyles-root-16 makeStyles-icon-7"
              >
                gem-solid.svg
              </svg>
            </div>
          </div>
        </div>
        <div
          className="makeStyles-button-5"
        >
          <div
            className="makeStyles-content-6"
          >
            <div
              aria-describedby={null}
              className=""
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              title="Help Desk"
            >
              <svg
                className="makeStyles-root-10 makeStyles-root-17 makeStyles-icon-7"
              >
                tools.svg
              </svg>
            </div>
          </div>
        </div>
        <div
          className="makeStyles-button-5"
        >
          <div
            className="makeStyles-content-6"
          >
            <div
              aria-describedby={null}
              className=""
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              title="Trocar Conta"
            >
              <svg
                className="makeStyles-root-10 makeStyles-root-18 makeStyles-icon-7"
              >
                exchange-alt.svg
              </svg>
            </div>
          </div>
        </div>
        <div
          className="makeStyles-button-5"
        >
          <div
            className="makeStyles-content-6"
          >
            <div
              aria-describedby={null}
              className=""
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              title="Sair"
            >
              <svg
                className="makeStyles-root-10 makeStyles-root-19 makeStyles-icon-7"
              >
                sign-out-alt.svg
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
